lib/rules/indent.js

```diff
@@ -192,108 +190,85 @@ module.exports = {
-                        rangeToFix = [
-                            node.range[1] - 1,
-                            node.range[1] - 1
-                        ];
-                    } else {
-                        rangeToFix = [
-                            node.range[0],
-                            node.range[0]
-                        ];
-                    }
+        function report(node, needed, gottenSpaces, gottenTabs, loc, isLastNodeCheck) {
 
-                    return function(fixer) {
-                        return fixer.insertTextBeforeRange(rangeToFix, spaces);
-                    };
-                } else {
-                    if (isLastNodeCheck === true) {
-                        rangeToFix = [
-                            node.range[1] - (gotten - needed) - 1,
-                            node.range[1] - 1
-                        ];
-                    } else {
-                        rangeToFix = [
-                            node.range[0] - (gotten - needed),
-                            node.range[0]
-                        ];
-                    }
+            const desiredIndent = (indentType === "space" ? " " : "\t").repeat(needed);
 
-                    return function(fixer) {
-                        return fixer.removeRange(rangeToFix);
-                    };
-                }
-            }
+            const textRange = isLastNodeCheck
+                ? [node.range[1] - gottenSpaces - gottenTabs - 1, node.range[1] - 1]
+                : [node.range[0] - gottenSpaces - gottenTabs, node.range[0]];
 
-            if (loc) {
-                context.report({
-                    node,
-                    loc,
-                    message: MESSAGE,
-                    data: msgContext,
-                    fix: getFixerFunction()
-                });
-            } else {
-                context.report({
-                    node,
-                    message: MESSAGE,
-                    data: msgContext,
-                    fix: getFixerFunction()
-                });
-            }
+            context.report({
+                node,
+                loc,
+                message: createErrorMessage(needed, gottenSpaces, gottenTabs),
+                fix: fixer => fixer.replaceTextRange(textRange, desiredIndent)
+            });
         }
 
         /**
          * Get the actual indent of node
          * @param {ASTNode|Token} node Node to examine
          * @param {boolean} [byLastLine=false] get indent of node's last line
          * @param {boolean} [excludeCommas=false] skip comma on start of line
-         * @returns {int} Indent
+         * @returns {Object} The node's indent. Contains keys `space` and `tab`, representing the indent of each character. Also
+         contains keys `goodChar` and `badChar`, where `goodChar` is the amount of the user's desired indentation character, and
+         `badChar` is the amount of the other indentation character.
```

Comment:
It seems to me that spaces are always considered "goodChar".
cf https://github.com/eslint/eslint/pull/7076/commits/de9d9e7c133b49a8bbd26d156cf156268aa68c07#diff-f7ae198ea24182d178b8f3376e61b6edR270